var/spack/repos/builtin/packages/gnuplot/package.py

```diff
@@ -43,24 +41,97 @@ class Gnuplot(Package):
     homepage = "http://www.gnuplot.info"
     url      = "http://downloads.sourceforge.net/project/gnuplot/gnuplot/5.0.1/gnuplot-5.0.1.tar.gz"
 
+    # There is a conflict in term.h between gnuplot and ncurses, which is a
+    # dependency of readline. Fix it with a small patch
+    patch('term_include.patch')
+
+    version('5.0.5', 'c5e96fca73afbee4f57cbc1bfce6b3b8')
     version('5.0.1', '79b4f9e203728f76b60b28bcd402d3c7')
 
+    variant('wx',      default=False,
+            description='Activates wxWidgets terminal')
+    variant('gd',      default=True,
+            description='Activates gd based terminal')
+    variant('cairo',   default=True,
+            description='Activates cairo based terminal')
+    variant('X',       default=False,
+            description='Build with X11')
+    variant('libcerf', default=True,
+            description='Build with libcerf support')
+    variant('pbm',     default=False,
+            description='Enable PBM (Portable Bit Map) and other older bitmap terminals')  # NOQA: ignore=E501
+
+    # required dependencies
     depends_on('readline')
-    depends_on('libcerf')
-    depends_on('libgd')
-    depends_on('cairo')
-    depends_on('pango')
+
+    # optional dependencies:
+    depends_on('libcerf', when='+libcerf')
+    depends_on('libgd', when='+gd')
+    depends_on('cairo', when='+cairo')
     depends_on('wx', when='+wx')
+    depends_on('pango', when='+wx')
+    depends_on('pango', when='+cairo')
+    # FIXME: macports add this dependency
+    # depends_on('libxpm', when='+x11')
```

Comment:
I'm not sure if you have access to Linux, but if you build it on Linux, you can run `ldd -r` on the shared libraries/binaries to see whether or not it links to `libxpm`. I'm sure there is an equivalent for macOS.